golem-rib/src/function_name.rs

```diff
@@ -41,6 +43,26 @@ pub enum ParsedFunctionSite {
 #[derive(PartialEq, Eq, Clone)]
 pub struct SemVer(pub semver::Version);
 
+impl Serialize for SemVer {
```

Comment:
Where are we serializing this to json?

Reply:
This is re-used in the instruction set, and the instruction set (RibByteCode) is given `Serialize` and `Deserialise`.
However, I realise, why are we json serde RibByteCode. Let me see if I can remove this requirement.

Reply:
Yeah let's not have unnecessary derived serde instances, it just slows compilation down

Reply:
Removed